spring-integration-webflux/src/main/java/org/springframework/integration/webflux/outbound/WebFluxRequestExecutingMessageHandler.java

```diff
@@ -198,6 +202,26 @@ public void setPublisherElementTypeExpression(Expression publisherElementTypeExp
 		this.publisherElementTypeExpression = publisherElementTypeExpression;
 	}
 
+	/**
+	 * Specify the timeout value for receiving the response form the server.
+	 * If the response is not received with in timeout value, will result in Timeout Exception
+	 * @param timeout accepts {@link java.time.Duration}
+	 * @since 5.3
+	 */
+	public void setTimeout(Duration timeout) {
+		this.timeoutFunction = m -> timeout;
+	}
+
+	/**
+	 * @param timeoutFunction accepts {@link Function}. The function accepts a Message as input and returns Duration.
+	 * Function is evaluated on each request basis and the {@link Duration} is applied to Mono
+	 * If the response is not received with in timeout value, will result in Timeout Exception
+	 * @since 5.3
+	 */
+	public void setTimeoutFunction(Function<Message<?>, Duration> timeoutFunction) {
```

Comment:
Well, we prefer an `Expression` variant instead of function. This way a timeout could be configured in the XML an easy way.
There is a `FunctionExpression` to cover this case. And exactly Java DSL should expose a function-based option. Of course, alongside with an expression. See `publisherElementTypeFunction` in the modified by you `WebFluxMessageHandlerSpec`